pkg/kwok/metrics/cel/evaluate.go

```diff
@@ -0,0 +1,211 @@
+	env            *easycel.Environment
+	conf           NodeEvaluatorConfig
+	cacheEvaluator map[string]*Evaluator
+}
+
+func (e *Environment) init() error {
+	conversions := []any{
+		func(t metav1.Time) types.Timestamp {
+			return types.Timestamp{Time: t.Time}
+		},
+	}
+	types := []any{
+		corev1.Node{},
+		metav1.ObjectMeta{},
+	}
+
+	vars := map[string]any{
+		"node": corev1.Node{},
+	}
+
+	funcs := map[string][]any{}
+
+	methods := map[string][]any{}
+
+	if e.conf.Now != nil {
+		funcs["now"] = append(funcs["now"], e.conf.Now)
+	} else {
+		funcs["now"] = append(funcs["now"], time.Now)
+	}
+
+	unixSecond := func(t time.Time) int64 {
+		return t.Unix()
+	}
+	methods["unixSecond"] = append(methods["unixSecond"], unixSecond)
+	funcs["unixSecond"] = append(funcs["unixSecond"], unixSecond)
+
+	if e.conf.StartedContainersTotal != nil {
+		startedContainersTotal := e.conf.StartedContainersTotal
+		startedContainersTotalByNode := func(node corev1.Node) int64 {
+			return e.conf.StartedContainersTotal(node.Name)
+		}
+		methods["startedContainersTotal"] = append(methods["startedContainersTotal"], startedContainersTotal, startedContainersTotalByNode)
+		funcs["startedContainersTotal"] = append(funcs["startedContainersTotal"], startedContainersTotal, startedContainersTotalByNode)
+	}
+
+	for _, convert := range conversions {
+		err := e.registry.RegisterConversion(convert)
+		if err != nil {
+			return fmt.Errorf("failed to register convert %T: %w", convert, err)
+		}
+	}
+	for _, typ := range types {
+		err := e.registry.RegisterType(typ)
+		if err != nil {
+			return fmt.Errorf("failed to register type %T: %w", typ, err)
+		}
+	}
+	for name, val := range vars {
+		err := e.registry.RegisterVariable(name, val)
+		if err != nil {
+			return fmt.Errorf("failed to register variable %s: %w", name, err)
+		}
+	}
+	for name, list := range funcs {
+		for _, fun := range list {
+			err := e.registry.RegisterFunction(name, fun)
+			if err != nil {
+				return fmt.Errorf("failed to register function %s: %w", name, err)
+			}
+		}
+	}
+	for name, list := range methods {
+		for _, fun := range list {
+			err := e.registry.RegisterMethod(name, fun)
+			if err != nil {
+				return fmt.Errorf("failed to register method %s: %w", name, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// Compile is responsible for compiling a cel program
+func (e *Environment) Compile(src string) (*Evaluator, error) {
+	if evaluator, ok := e.cacheEvaluator[src]; ok {
+		return evaluator, nil
+	}
+	program, err := e.env.Program(src)
+	if err != nil {
+		return nil, fmt.Errorf("failed to compile metric expression: %w", err)
+	}
+
+	evaluator := &Evaluator{
+		program: program,
+	}
+	e.cacheEvaluator[src] = evaluator
+	return evaluator, nil
+}
+
+// Evaluator evaluates a cel program
+type Evaluator struct {
+	program cel.Program
+}
+
+// EvaluateMetric evaluates a cel program and returns a metric value
+func (e *Evaluator) EvaluateMetric(node *corev1.Node) (float64, error) {
```

Comment:
```suggestion
func (e *Evaluator) EvaluateFloat64(node *corev1.Node) (float64, error) {
```

Consistent with the style of `EvaluateString`?